cpp/src/codegen/tree_expr_builder.cc

```diff
@@ -37,6 +37,14 @@ MAKE_LITERAL(arrow::uint64(), uint64_t)
 MAKE_LITERAL(arrow::float32(), float)
 MAKE_LITERAL(arrow::float64(), double)
 
+NodePtr TreeExprBuilder::MakeStringLiteral(const std::string &value) {
+  return std::make_shared<LiteralNode>(arrow::utf8(), LiteralHolder(value), false);
+}
+
+NodePtr TreeExprBuilder::MakeBinaryLiteral(const std::string &value) {
+  return std::make_shared<LiteralNode>(arrow::binary(), LiteralHolder(value), false);
```

Comment:
does holder make a copy of the input? if not should this not be a ref..

Reply:
yes. The list of types defined in the template for LiteralHolder (in literal_holder.h) has std::string. so, it must do a copy.